internal/charmstore/search.go

```diff
@@ -83,10 +83,7 @@ func (s *Store) UpdateSearchAsync(r *router.ResolvedURL) {
 // The search index only includes the latest revision of each entity so
 // the latest revision of the charm specified by r will be indexed.
 func (s *Store) UpdateSearch(r *router.ResolvedURL) error {
-	if s.ES == nil || s.ES.Database == nil {
-		return nil
-	}
-	if deprecatedSeries[r.URL.Series] {
+	if s.ES == nil || s.ES.Database == nil || deprecatedSeries[r.URL.Series] || r.Development {
```

Comment:
Bike shedding here, but splitting the checks might be ok here. First one is state of the system, second one is internal logic for charms.